### PR TITLE
Homework4: Aliaksei Uhalnikau

### PR DESCRIPTION
**Pull request by alexugalek**

### CODE REVIEW

File: src/cities.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+import functools
+import time
+DEBUG = True
+
+
+def function_time_counter(func):
```
Comment (belovmd) [inline src/cities.py:6]: great

File: src/parser.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
+from bs4 import BeautifulSoup as bsoup
```
Comment (belovmd) [inline src/parser.py:1]: nice

File: src/task_1.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    :param name: A function name as a string
+    :return: The same string, but in CamelCase
+    """
+    new_string = ''
+    for index, char in enumerate(name):
+        if index == 0:
+            new_string += char.upper()
+        elif name[index - 1] == '_':
+            new_string += char.upper()
+        elif char != '_':
```
Comment (belovmd) [inline src/task_1.py:17]: else?

File: src/task_5.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    The status string should be set to: "same" if f and g return the same
+    output and neither errs, "different" if f and g return different outputs
+    and neither errs, "f_error" if f errs but not g, "g_error" if g errs but
+    not f, or "both_error" if both err.
+
+    :param f: functions f (primary)
+    :param g: functions g (back up function)
+    :return: A function h which takes arbitrary inputs and returns a two-tuple
+    """
+    def h(*args, **kwargs):
```
Comment (belovmd) [inline src/task_5.py:24]: 1. looks too complex but i don't see quick win for this for now
2. what if g returns None but doesn't raise exception?

**Review by belovmd — CHANGES_REQUESTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-python-spring/blob/master/homework_review/Homework4.md):
- 14
- I haven't understood your file names :)
- please address next issues:

File: src/task_5.py (hunk 4 above)
Comment (alexugalek) [reply src/task_5.py:24]: 2. what if g returns None but doesn't raise exception?

if f_result is not None:
    if g_result is None:
                    return f_result, 'g_error'

if f_result == g_result and f_result is None:
            return None, 'both_error'

**Review by belovmd — COMMENTED** [review]
9,5

**Pull request closed without merging**